### PR TITLE
Delete conference when hearing is postponed or cancelled

### LINKED ISSUE #15447: Bug: When postponing a hearing, the virtual hearing does not get cancelled
## Description

Previously a virtual hearing was only cancelled when the hearing switched types back to the original type (e.g. switching a virtual hearing back to video would cancel the virtual hearing).

Now that you can postpone and reschedule a virtual hearing, we need to cancel the associated virtual hearing (set `request_cancelled` and start the delete conference job) when the hearing is postponed.

## Acceptance criteria

- [ ] Ensure that when a virtual hearing is postponed, the:
  - [ ] Delete conference job runs
  - [ ] `request_cancelled` is set to true on the virtual hearing
- [ ] Cleanup data in production

## Background/context/resources

See: `LegacyAppeal.find(1019872)`

https://github.com/department-of-veterans-affairs/caseflow/blob/3fedcf6c4f7bf792d94bfb91e3161c51917cbecf/app/models/tasks/assign_hearing_disposition_task.rb#L185

### PR DESCRIPTION
**Pull request by rubaiyat22**
Resolves #15447

### Description
- Ensure `VirtualHearingRepository` considers hearing that have disposition of `cancelled` or `postponed`
- Ensure `DeleteConferencesJob` does not send cancellation emails if hearing is `postponed` or `cancelled`
- Write tests

### Acceptance Criteria
- [x] Ensure that when a virtual hearing is postponed, the:
  - [x] Delete conference job runs
  - ~[ ] `request_cancelled` is set to true on the virtual hearing~
- [ ] Cleanup data in production

### CODE REVIEW

File: app/jobs/virtual_hearings/delete_conferences_job.rb [hunk 1]
```diff
@@ -87,8 +87,16 @@ def ensure_current_user_is_set
     RequestStore.store[:current_user] ||= User.system_user
   end
 
+  def hearing_is_postponed_or_cancelled?(virtual_hearing)
```
Comment (rubaiyat22) [inline app/jobs/virtual_hearings/delete_conferences_job.rb:90]: Prevent sending cancellation emails if hearing is postponed or cancelled

File: app/models/hearings/virtual_hearing.rb [hunk 2]
```diff
@@ -13,7 +13,24 @@
 # emails to participants including the link to video conference as well as other details about
 # the hearing. DeleteConferencesJob is kicked off to delete the conference resource
 # when the the virtual hearing is cancelled or after the hearing takes place.
-
+#
+# A note about `status`:
```
Comment (ferristseng) [inline app/models/hearings/virtual_hearing.rb:17]: Amazing explanation of all the different statuses, thank you!!

File: app/repositories/virtual_hearing_repository.rb [hunk 3]
```diff
@@ -8,17 +8,20 @@ def ready_for_deletion
...
 
       virtual_hearings_for_legacy_hearings = VirtualHearing.eligible_for_deletion
         .where(hearing_type: LegacyHearing.name)
         .joins("INNER JOIN legacy_hearings ON legacy_hearings.id = virtual_hearings.hearing_id")
         .joins("INNER JOIN hearing_days ON hearing_days.id = legacy_hearings.hearing_day_id")
         .where(
-          "hearing_days.scheduled_for < :today OR virtual_hearings.request_cancelled = true",
-          today: Time.zone.today
+          "hearing_days.scheduled_for < :today OR
+          legacy_hearings.vacols_id in (:postponed_or_cancelled_vacols_ids) OR
```
Comment (ferristseng) [inline app/repositories/virtual_hearing_repository.rb:22]: Nice!

File: app/models/legacy_hearing.rb [hunk 4]
```diff
@@ -188,6 +188,14 @@ def scheduled_pending?
     scheduled_for && !closed?
   end
 
+  def postponed?
+    disposition == Constants.HEARING_DISPOSITION_TYPES.postponed
+  end
+
+  def cancelled?
```
Comment (ferristseng) [inline app/models/legacy_hearing.rb:195]: One minor suggestion is to move these checks into a concern that we can share between both the `LegacyHearing` and `Hearing` classes to make the code slightly more DRY.

**Review by ferristseng — APPROVED** [review]
This looks amazing! Thank you for adding the detailed documentation about statuses!

I made one small suggestion, but otherwise feel free to 🚢 !

**Pull request merged**